src/units/abilities.cpp

```diff
@@ -1168,33 +1168,89 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
-//Functions used for change damage_type list with damage
-static std::optional<std::string> select_damage_type(const unit_ability_list& abil_list, const std::string& type)
+//functions for select damage type who must be returned for each attribute used in damage_type()
+static std::optional<std::string> select_replacement_type_or_alternative_without_resistance(const unit_ability_list& abil_list, const std::string& key_name)
```

Comment:
What are the cases for alternative_type without resistance? Is that the report thing again?

Reply:
Yes

Reply:
Ok, as mentioned in the other comment I would much rather get rid of that. If you don't want to deal with that then I'd prefer to remove the logic from this function and check in the caller and return an empty string for now.